pkg/event/event.go

```diff
@@ -265,22 +303,17 @@ connect:
 				//TODO: update clock class is going to be called for every event, this is not efficient
 				// This is going to choke the system if there are too many events
 				// since pmc call takes time, have seen 5 seconds timeout error in logs
-				err, clockClass := e.updateCLockClass(event.CfgName, gmState, event.ClockType)
-				if err == nil {
-					// in case an error do not update lastGmState so that updateCLockClass can be tried again
-					lastGmState = gmState
-					logOut = append(logOut, fmt.Sprintf("%s[%d]:[%s] CLOCK_CLASS_CHANGE %d\n", PTP4l, time.Now().Unix(), event.CfgName, clockClass))
-				} else {
-					glog.Errorf("failed to updateClockClass: %v", err)
-				}
+				clockClassRequestCh <- ClockClassRequest{cfgName: event.CfgName, gmState: gmState, clockType: event.ClockType}
+				// in case an error do not update lastGmState so that updateCLockClass can be tried again
+				lastGmState = gmState
```

Comment:
The lastGmState should only be updated when updateClockClass() was successful, otherwise if updateClockClass() failed the function will not be triggered again.